File: backend/src/models/notes.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Note {
    pub id: String,
    pub title: String,
    pub content: String, // Markdown
    pub created_by: String,
    pub updated_by: String,
    #[serde(default)]
    pub shared_with: Vec<String>,
    #[serde(default)]
    pub is_public: bool,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

File: frontend/src/types/notes.ts
export interface Note {
  id: string
  title: string
  content: string
  created_by: string
  updated_by: string
  shared_with: string[]
  is_public: boolean
  created_at: string
  updated_at: string
}

export interface ShareLink {
  token: string
  file_name: string
  file_path: string
  url: string
  remaining_minutes: number
}
